### ISSUE TITLE
[Filecoin] Support new f4 address types

### ISSUE THREAD

**Issue by hewigovens** [trustwallet/wallet-core#3008]
As part of the Filecoin Virtual Machine’s EVM-compatibility, Filecoin has now added 2 address types:
- 0x addresses that are mapped to
- f4 addresses

More info about these new address types:
- [Exchange Integration Guide - FEVM Edition](https://www.notion.so/pl-strflt/Exchange-Integration-Guide-FEVM-Edition-9fe9158360cf423da256b1bb1f47ff26#3ffd228450f24beebf7459f53f82c61c)
- [Filecoin Docs - Wallets section](https://docs.filecoin.io/smart-contracts/wallets/transfer-fil/)

TODOs:

- [ ] validate `f4` and `0x` addresses
- [ ] method mapping existed `f1` to `f410` ETH address
- [ ] add sending to `f4` / `0x` address tx test

**Comment by Milerius** [comment]
For 0x address, simply list the Filecoin EVM chain.

add function for converting F4 to 0x and vice-versa + expose it to FFI.

**Comment by Milerius** [comment]
It should be possible to extend AnyAddress interface for that @satoshiotomakan - we've done similar approach for SS58 and Bech32 in the past @hewigovens - what do you think?

**Comment by hewigovens** [comment]
The minimum is to support sending to f4 and 0x addresses so it's ok not to support mapping from f1 to 0x (better to confirm with filecoin team, I added you to the channel)

Try to avoid changing registry, Roman's approach looks good

**Comment by Milerius** [comment]
> The minimum is to support sending to f4 and 0x addresses so it's ok not to support mapping from f1 to 0x (better to confirm with filecoin team, I added you to the channel)
> 
> Try to avoid changing registry, Roman's approach looks good

after discussing it's simple as:

- Support f4
- Add conversion F4/0x and vice versa
- Expose conversion functions
- End user app use existing interface to validate: Eth::isValid(f4to0x(address)) and so on.

**Comment by satoshiotomakan** [comment]
If we want to allow the end user to generate both `f1` and `f4` addresses, we also should support sending transactions from the new `f4` addr.
The signature type in that case is different - `type 3` called [Delegated](https://github.com/filecoin-project/go-state-types/blob/master/crypto/signature.go#L19).
As I could find in [Lotus codebase](https://github.com/filecoin-project/lotus), first the Filecoin message is converted to an Ethereum transaction, [encodes to rlp bytes](https://github.com/filecoin-project/lotus/blob/master/chain/messagesigner/messagesigner.go#L197-L211), and then its hash is signed (the same way as for EIP1559 transaction).

The problem is that such approach doesn't allow to transfer funds directly from `f4` address to `f1`/`f2`/`f3`, but using a "forward" method: https://github.com/filecoin-project/lotus/blob/5f91072d4d7d57a2943a4bb396b2986b57597190/chain/types/ethtypes/eth_transactions.go#L132-L146

Please take a look at this [transaction](https://explorer.glif.io/tx/bafy2bzacecjrexs2no7tn3hcu5l3fwcgelbaq3ekyrlutdeefdx3pqr52hhte/?network=hyperspacenet) I've sent from `t410frw6wy7w6sbsguyn3yzeygg34fgf72n5ao5sxyky` to `t1mzyorxlcvdoqn5cto7urefbucugrcxxghpjc5hi`. The payload contains an Smart Contract Call (I believe), and the actual message's recipient is [t410ffm7pnedefg2ybn5sbag6lsujhpbifqrfspbcqna](https://explorer.glif.io/address/t410ffm7pnedefg2ybn5sbag6lsujhpbifqrfspbcqna/?network=hyperspacenet) (Actor Name = Evm).

To sum up, we (as Wallet Core) should allow the end user to send messages in the following scheme:
f1, f2, f3 -> f1, f2, f3, f4
f4 -> f4

My question is should I just return an empty `Proto::SigningOutput` if the user tries to sign `f4` -> `f1/f2/f3` transaciton?
cc: @hewigovens @Milerius
